### PR TITLE
Upgrade to pulsar client 2.10.2

### PR DESCRIPTION
**Pull request by cbornet**
Pulsar 2.10.2 makes `ConsumerConfigurationData.deadLetterPolicy` serializable so we can simplify `ConsumerBuilderConfigurationUtil::loadConf`

### CODE REVIEW

File: spring-pulsar/src/main/java/org/springframework/pulsar/core/ConsumerBuilderConfigurationUtil.java [hunk 1]
```diff
@@ -46,52 +43,46 @@ private ConsumerBuilderConfigurationUtil() {
...
 	 * @param <T> the payload type
 	 */
-	@SuppressWarnings("unchecked")
 	public static <T> void loadConf(ConsumerBuilder<T> builder, Map<String, Object> properties) {
 
 		ConsumerConfigurationData<T> builderConf = ((ConsumerBuilderImpl<T>) builder).getConf();
 		Map<String, Object> propertiesCopy = new HashMap<>(properties);
 
-		// Remove and remember problem fields from input props and builder
-		DeadLetterPolicy deadLetterPolicy = getValueToApplyToBuilderAfterLoadConf(builderConf::getDeadLetterPolicy,
```
Comment (onobc) [inline spring-pulsar/src/main/java/org/springframework/pulsar/core/ConsumerBuilderConfigurationUtil.java:61]: Nice improvement @cbornet . Thank you for getting the fix in for the PulsarClient! Do you mind updating the [unit test](https://github.com/spring-projects-experimental/spring-pulsar/blob/bb12a7b666c59af178571eba47bb81e35732c6e1/spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerBuilderConfigurationUtilTests.java#L84) to include the newly added fields?

**Review by onobc — CHANGES_REQUESTED** [review]
Nice improvement @cbornet . Thank you for getting the fix in for the PulsarClient! Do you mind updating the [unit test](https://github.com/spring-projects-experimental/spring-pulsar/blob/bb12a7b666c59af178571eba47bb81e35732c6e1/spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerBuilderConfigurationUtilTests.java#L84) to include the newly added fields?

**Comment by cbornet** [comment]
Sure, I'll update the UT.
Also note that tests seem to fail with Pulsar client 2.10.2... Maybe there's some kind of regression in Pulsar 😞

File: spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerTestUtils.java [hunk 2]
```diff
@@ -35,11 +45,20 @@ private ConsumerTestUtils() {
 	 * @param container container to spy on
 	 * @return the spied container object
 	 */
-	public static Consumer<?> spyOnConsumer(DefaultPulsarMessageListenerContainer<String> container) {
-		Consumer<?> consumer = getPropertyValue(container, "listenerConsumer.consumer", Consumer.class);
-		consumer = spy(consumer);
-		new DirectFieldAccessor(getPropertyValue(container, "listenerConsumer")).setPropertyValue("consumer", consumer);
-		return consumer;
+	@SuppressWarnings("unchecked")
+	public static Consumer<String> startContainerAndSpyOnConsumer(
```
Comment (cbornet) [inline spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerTestUtils.java:49]: For some reason, when upgrading to 2.10.2, the former spyOnConsumer was resulting in a deadlock in `Consumer::batchReceive`.
This is an alternative that works better.

File: spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerTestUtils.java [hunk 3]
```diff
@@ -35,11 +45,20 @@ private ConsumerTestUtils() {
...
+			throws PulsarClientException, ExecutionException, InterruptedException, TimeoutException {
+		CompletableFuture<Consumer<String>> consumerFuture = new CompletableFuture<>();
+		doAnswer(invocation -> {
+			Consumer<String> consumer = spy((Consumer<String>) invocation.callRealMethod());
+			consumerFuture.complete(consumer);
+			return consumer;
+		}).when(container.getPulsarConsumerFactory()).createConsumer(any(), isNull(), isNull(), anyList());
+
+		container.start();
+		Thread.sleep(1000);
```
Comment (cbornet) [inline spring-pulsar/src/test/java/org/springframework/pulsar/core/ConsumerTestUtils.java:60]: We need to give some time for the consumer to initialise.

**Comment by cbornet** [comment]
@onobc I did the changes. PTAL.

**Review by onobc — APPROVED** [review]
LGTM @cbornet . Thanks for the updates.

**Comment by onobc** [comment]
Closed via https://github.com/spring-projects-experimental/spring-pulsar/commit/7e91f78fa1fff530e172d5bad450a323973b0d24

**Pull request closed without merging**